imagebuildah/stage_executor.go

```diff
@@ -1164,6 +1173,24 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 		// determining if a cached layer with the same build args already exists
 		// and that is done in the if block below.
 		if checkForLayers && step.Command != "arg" && !(s.executor.squash && lastInstruction && lastStage) {
+			// Is true if current STEP is `COPY` or `ADD`, denoting that
+			// cache lookup can be attempted only after performing a RUN
```

Comment:
```suggestion
			// For `COPY` and `ADD`, history entries include digests computed from
			// the content that's copied in.  We need to compute that information so that
			// it can be used to evaluate the cache, which means we need to go ahead
			// and copy the content.
```